src/parser/parser.rs

```diff
@@ -11,99 +11,119 @@ use serde_json::Value;
 #[grammar = "parser/grammar/json_path.pest"]
 struct JsonPathParser;
 
-/// the parsing function.
-/// Since the parsing can finish with error the result is [[Result]]
-#[allow(clippy::result_large_err)]
-pub fn parse_json_path(jp_str: &str) -> Result<JsonPath, Error<Rule>> {
-    Ok(parse_internal(
-        JsonPathParser::parse(Rule::path, jp_str)?.next().unwrap(),
-    ))
+/// Parses a string into a [JsonPath].
+///
+/// # Errors
+///
+/// Returns a variant of [JsonPathParserError] if the parsing operation failed.
+pub fn parse_json_path(jp_str: &str) -> Result<JsonPath, JsonPathParserError> {
+    match JsonPathParser::parse(Rule::path, jp_str)?.next() {
+        Some(parsed_pair) => Ok(parse_internal(parsed_pair)?),
+        None => Err(JsonPathParserError::ParserError(format!(
+            "Failed to parse JSONPath {jp_str}"
+        ))),
+    }
 }
 
 /// Internal function takes care of the logic by parsing the operators and unrolling the string into the final result.
-fn parse_internal(rule: Pair<Rule>) -> JsonPath {
+///
+/// # Errors
+///
+/// Returns a variant of [JsonPathParserError] if the parsing operation failed
+fn parse_internal(rule: Pair<Rule>) -> Result<JsonPath, JsonPathParserError> {
     match rule.as_rule() {
-        Rule::path => rule
+        Rule::path => Ok(rule
             .into_inner()
             .next()
             .map(parse_internal)
-            .unwrap_or(JsonPath::Empty),
-        Rule::current => JsonPath::Current(Box::new(
+            .unwrap_or(Ok(JsonPath::Empty))
```

Comment:
Basically, since we have `Result` as a result we don't have to return JsonPath::Empty and can return Err instead (apparently). I would suggest replacing unwrap_or everywhere to ok_or(Err(..))